### PR TITLE
CI: Added integration tests for eos_designs_structured_config plugin action

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Added intergration tests for eos_designs_structured_config plugin action.

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.<role-name>`

## Proposed changes
Added intergration tests for eos_designs_structured_config plugin action.

## How to test
Check the coverage for eos_designs_structured_config plugin action
1. ansible-test coverage erase
2. ansible-test integration eos_designs_structured_config --python 3.12 --coverage
3. ansible-test coverage combine --python 3.12
4. ansible-test coverage report --python 3.12 --include 'plugins/action/eos_designs_structured_config.py' --show-missing

Coverage from devel
```
(avd_env) mahesh.kumar-ext@mahesh49RVC7m avd % ansible-test coverage report --python 3.12 --include 'plugins/action/eos_designs_structured_config.py' --show-missing
Name                                              Stmts   Miss Branch BrPart  Cover   Missing
---------------------------------------------------------------------------------------------
plugins/action/eos_designs_structured_config.py      89     11     34      7    85%   42-43, 53-54, 60, 115, 125, 129->132, 176-180, 205-206
---------------------------------------------------------------------------------------------
TOTAL                                                89     11     34      7    85%
(avd_env) mahesh.kumar-ext@mahesh49RVC7m avd % 
```

Coverage with this PR
```
(avd_env) mahesh.kumar-ext@mahesh49RVC7m avd % ansible-test coverage report --python 3.12 --include 'plugins/action/eos_designs_structured_config.py' --show-missing
Name                                              Stmts   Miss Branch BrPart  Cover   Missing
---------------------------------------------------------------------------------------------
plugins/action/eos_designs_structured_config.py      89      4     34      1    96%   42-43, 53-54
---------------------------------------------------------------------------------------------
TOTAL                                                89      4     34      1    96%
```

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.arista.com/stable/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/extensions/molecule) testing accordingly. (check the box if not applicable)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of custom template outputs, including empty, null, list, and boolean values.
  * Clarified error reporting when required design facts or validated inputs are missing.
  * Confirmed that disabling structured configuration completes without changes or generated facts.

* **Tests**
  * Added integration coverage for custom template edge cases, missing prerequisites, disabled structured configuration, and empty template results.
  * Strengthened validation of expected error messages and invalid input handling.

### CODE REVIEW

File: ansible_collections/arista/avd/tests/integration/targets/eos_designs_structured_config/tests/test_with_empty_custom_template_output.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/tests/integration/targets/eos_designs_structured_config/tests/test_with_empty_custom_template_output.yml:1]: I think this test can be combined with custom_template_edge_cases, can you please try combining them

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/tests/integration/targets/eos_designs_structured_config/tests/test_with_empty_custom_template_output.yml:1]: we can but we kept separate test for each scenario. If we need to combined then better combined all all files into single one.

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/tests/integration/targets/eos_designs_structured_config/tests/test_with_empty_custom_template_output.yml:1]: Checked the existing tests, to be consistent lets keep the tests sepaartely

File: ansible_collections/arista/avd/tests/integration/targets/eos_designs_structured_config/templates/false_output.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{#
+ Copyright (c) 2026 Arista Networks, Inc.
+ Use of this source code is governed by the Apache License 2.0
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/tests/integration/targets/eos_designs_structured_config/templates/false_output.yml.j2:3]: I think this template should be named false_output.yml

**Review by Shivani-gslab — APPROVED** [review]
just a minor comment, rest LGTM

File: ansible_collections/arista/avd/tests/integration/targets/eos_designs_structured_config/templates/false_output.yml.j2 (hunk 2 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/tests/integration/targets/eos_designs_structured_config/templates/false_output.yml.j2:3]: updated

**Review by laxmikantchintakindi — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
